### PR TITLE
WIP: amp-sticky-ad swipe dismiss feature

### PR DESCRIPTION
**Pull request by zhouyx**
#2472

### CODE REVIEW

**Comment by sriramkrish85** [comment]
@zhouyx ill review this tomorrow. We can sit together and review.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 1]
```diff
@@ -50,6 +51,12 @@ class AmpStickyAd extends AMP.BaseElement {
     /** @private @const {!Viewport} */
     this.viewport_ = this.getViewport();
 
+    /** @private @const {!Vsync} */
+    this.vsync_ = this.getVsync();
+
+    /** @private @const {!Function|null}*/
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: Not const.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 2]
```diff
@@ -84,6 +92,14 @@ class AmpStickyAd extends AMP.BaseElement {
     }
   }
 
+  removeOnHorizontalListener_() {
+    if (this.bindedOnHorizontalScroll_) {
+      this.element.removeEventListener('scroll',
+          this.bindedOnHorizontalScroll_);
+      this.bindedOnHorizontalScroll_ = null;
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: Is there a reason to throw away the listener? (Not unbind it -- we definitely should do that. But why throw away the reference to the bound function)

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.js [hunk 3]
```diff
@@ -94,6 +110,7 @@ class AmpStickyAd extends AMP.BaseElement {
     const scrollTop = this.viewport_.getScrollTop();
     const viewportHeight = this.viewport_.getSize().height;
     const scrollHeight = this.viewport_.getScrollHeight();
+    this.viewportWidth_ = this.viewport_.getSize().width;
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.js]: There's `Viewport#getWidth`.

**Pull request closed without merging**